bucky/carbon.py

```diff
@@ -65,19 +65,19 @@ def connect(self):
             self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             try:
                 self.sock.connect((self.ip, self.port))
-                log.info("Connected to Carbon at %s:%s", self.ip, self.port)
+                log.info("Connected to Carbon at %s:%s" % (self.ip, self.port))
                 return
             except socket.error as e:
                 if i >= self.max_reconnects:
                     raise
-                log.error("Failed to connect to %s:%s: %s", self.ip, self.port, e)
+                log.error("Failed to connect to %s:%s: %s" % (self.ip, self.port, e))
                 if reconnect_delay > 0:
                     time.sleep(reconnect_delay)
                     if self.backoff_factor:
                         reconnect_delay *= self.backoff_factor
                         if self.backoff_max:
                             reconnect_delay = min(reconnect_delay, self.backoff_max)
-        raise socket.error("Failed to connect to %s:%s after %s attempts", self.ip, self.port, self.max_reconnects)
+        raise socket.error("Failed to connect to %s:%s after %s attempts" % (self.ip, self.port, self.max_reconnects))
```

Comment:
As far as I can tell, this is the only line from the diff that should be merged.